hail/src/main/scala/is/hail/io/fs/AzureStorageFS.scala

```diff
@@ -66,53 +66,34 @@ class AzureStorageFSHttpsURL(
 
 
 object AzureStorageFS {
-  private val pathRegex = "/([^/]+)(.*)".r
+  private val HAIL_AZ_URI_REGEX = "^hail-az:\\/\\/([a-z0-9_\\-\\.]+)\\/([a-z0-9_\\-\\.]+)(\\/.*)?".r
+  private val AZURE_HTTPS_URI_REGEX = "^https:\\/\\/([a-z0-9_\\-\\.]+)\\.blob\\.core\\.windows\\.net\\/([a-z0-9_\\-\\.]+)(\\/.*)?".r
 
   private val log = Logger.getLogger(getClass.getName)
 
   val schemes: Array[String] = Array("hail-az", "https")
 
   def parseUrl(filename: String): AzureStorageFSURL = {
-    val uri = new URI(filename).normalize()
-
-    val scheme = uri.getScheme
-    if (scheme == null || !schemes.contains(scheme)) {
-      throw new IllegalArgumentException(s"Invalid scheme, expected hail-az or https: $scheme")
-    }
+    parseHttpsUrl(filename)
+      .orElse(parseHailAzUrl(filename))
+      .getOrElse(throw new IllegalArgumentException("ABS URI must be of the form https://<ACCOUNT>.blob.core.windows.net/<CONTAINER>/<PATH>"))
```

Comment:
Error reporting has gotten a little worse. Before we gave specialized errors depending on the scheme. Maybe it's worthwhile to still match the scheme first, then match the rest with the appropriate regex?